crates/apollo-compiler/src/validation/operation.rs

```diff
@@ -19,7 +19,13 @@ pub fn validate_operation_definitions(
         diagnostics
             .extend(db.validate_directives(def.directives().to_vec(), def.operation_ty().into()));
         diagnostics.extend(db.validate_variable_definitions(def.variables.as_ref().clone()));
-        diagnostics.extend(db.validate_selection_set(def.selection_set().clone()));
+        // TODO move this somewhere below the root operation type check
```

Comment:
ordering shouldn't really matter in validation 🤔 why does this need to be after root operation type check?